### DISCUSSION TITLE
Turbo Console Log

### DISCUSSION THREAD

**Discussion by maxpetretta** [zed-industries/zed#18671] [Feature Requests]
coming from the "other" editor, is there a way to replicate the functionality of the [Turbo Console Log](https://www.turboconsolelog.io/) extension? either with snippets or otherwise.

being able to quickly hover a variable and generate a log string on `ctrl-shift-l` is pretty ingrained for me at this point

**Comment by akickbush** [comment]
Got this to work with this via the keymap.json:

`[
  {
    "context": "Editor",
    "bindings": {
      "cmd-shift-l": [
        "workspace::SendKeystrokes",
        "cmd-c end enter console.log(\"value: \", cmd-v)"
      ]
    }
  }
]`

**Reply by maxpetretta** [reply]
thank you for pinging this! revisiting the question, i now use (in vim mode):

```
{
  "context": "Editor && (extension==js || extension==jsx || extension==ts || extension==tsx )",
  "bindings": {
    "cmd-shift-l": [
      "workspace::SendKeystrokes",
      "escape v i w y o console.log(`🚀 space ~ space cmd-v : space ${ cmd-v }`) escape"
    ]
  }
},
```

this gets me 95% of the way there

**Reply by matthiasbaldi** [reply]
I search exactly the same feature.
Thank you for the idea. But your explanation does not work for me.
```json
  {
    "context": "Editor",
    "bindings": {
      "ctrl-alt-l": [
        "workspace::SendKeystrokes",
        "ctrl-c end enter console.log('🚀 space ~ space ctrl-v :', ctrl-v) escape"
      ]
    }
  }
```

It prints only until the `,` (comma). the ctrl-v and the closing parenthesis is not printed. But the value is in the copy buffer, so a part works.
Do you have any idea why that is?
When I print it on one line like @maxpetretta did it, it works fine.

**Reply by r0skar** [reply]
@matthiasbaldi What worked for me (Linux 0.185.15) was to add an additional space between the `ctrl-v` expression and the parenthesis:

`ctrl-c end enter console.log( ctrl-v )`

**Reply by matthiasbaldi** [reply]
Ah thank you. I didn't try that out, but it works :smile: 

`"ctrl-c end enter console.log('🚀 space ~ space :', ctrl-v ) escape"`

**Reply by kostikovmu** [reply]
for me good

`"cmd-c end enter console.log(\"🚀 space cmd-v space ~ space :\", space cmd-v ); escape"`

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links

**Reply by tedlatentai** [reply]
Shoving everyone into this hellhole that [kills free access](https://www.reddit.com/r/LocalLLaMA/comments/1ao4aio/comment/kpwr64r/) to useful information is a damn stupid decision
